src/main/java/org/spongepowered/api/command/parameter/managed/ValueParameters.java

```diff
@@ -0,0 +1,502 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.command.parameter.managed;
+
+import com.flowpowered.math.vector.Vector3d;
+import org.spongepowered.api.CatalogType;
+import org.spongepowered.api.command.CommandSource;
+import org.spongepowered.api.command.parameter.managed.factories.ValueParameterFactory;
+import org.spongepowered.api.entity.Entity;
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.entity.living.player.User;
+import org.spongepowered.api.plugin.PluginContainer;
+import org.spongepowered.api.util.generator.dummy.DummyObjectProvider;
+import org.spongepowered.api.world.Dimension;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.storage.WorldProperties;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+
+/**
+ * Class containing common {@link ValueParameters}s.
+ */
+public final class ValueParameters {
+
+    private ValueParameters() {}
+
+    private static final ValueParameterFactory factory = DummyObjectProvider.createFor(ValueParameterFactory.class, "factory");
+
+    /**
+     * Require an argument to be a boolean.
+     *
+     * <p>The recognized true values are:</p>
+     *
+     * <ul>
+     *     <li>true</li>
+     *     <li>t</li>
+     *     <li>yes</li>
+     *     <li>y</li>
+     *     <li>veryumuchso</li>
+     * </ul>
+     *
+     * <p>The recognized false values are:</p>
+     *
+     * <ul>
+     *     <li>false</li>
+     *     <li>f</li>
+     *     <li>no</li>
+     *     <li>n</li>
+     *     <li>notatall</li>
+     * </ul>
+     *
+     * <p>Returns a {@link Boolean}</p>
+     */
+    public static ValueParameter bool() {
+        return CatalogedValueParameters.BOOLEAN;
+    };
+
+    /**
+     * Expect an argument to represent a dimension.
+     *
+     * <p>Returns a {@link Dimension}</p>
+     */
+    public static ValueParameter dimension() {
+        return CatalogedValueParameters.DIMENSION;
+    }
+
+    /**
+     * Require an argument to be an double-precision floating point number.
+     *
+     * <p>Returns a {@link Double}</p>
+     */
+    public static ValueParameter doubleNumber() {
```

Comment:
Given we’re breaking things, there is no need to keep current names. This will stay, other numbers will become <blah>Number, which I _had_ forgotten about for this file.